overlord/snapstate/handlers.go

```diff
@@ -975,7 +975,7 @@ func snapMonitoring(st *state.State) map[string]chan<- bool {
 }
 
 func abortMonitoring(st *state.State, snapName string) {
-	if monitored := snapMonitoring(st); monitored != nil {
+	if monitored := snapMonitoring(st); monitored != nil && monitored[snapName] != nil {
```

Comment:
nitpick: strictly you don't need the first "monitored != nil" because a nil map returns the zero value